src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -78,6 +78,11 @@ internal abstract partial class AbstractFlowPass<TLocalState, TLocalFunctionStat
         /// </summary>
         private readonly PooledDictionary<LabelSymbol, TLocalState> _labels;
 
+        /// <summary>
+        /// Ids used for sorting by label in PendingBranchesCollection.
+        /// </summary>
+        private PooledDictionary<LabelSymbol, int> _labelIds;
```

Comment:
>PooledDictionary<LabelSymbol, int> _labelIds [](http://example.com/codeflow?start=16&length=44)

This seems too heavyweight. Perhaps instead there should be a single `int index` that is incremented for each `PendingBranch` that is created analyzing the current method, and the `(int, PendingBranch)` are stored together in `PendingBranchesCollection`. Then `PendingBranchesCollection.AsEnumerable()` could return the branches in the order they were created.